docs/DeveloperGuide.md

```diff
@@ -1,38 +1,527 @@
+   `MenuUi#printError` and re-prompts the user to enter a new price.
+1. A new `Item` object is then created using the name and price given
+1. `Menu#appendItem` is invoked on the new `Item` object to add it to `ArrayList<Item> items` in `Menu`.
+1. The, `Menu#save` is invoked to save the changes to the local storage file.
+1. `Router#assistRoute` then calls `MenuAssistant#printResult` to print a message indicating that if the item has been 
+successfully added to the menu or if the user has cancelled the command accordingly.
+
+<hr>
+
+#### Delete an Item
+
+##### Advanced Mode Delete an Item
+The expected inputs to add only one menu item into an order is as such:
+* `/deleteitem -i <index>`
+
+This sequence diagram models the interaction between various components in MoneyGoWhere when the user inputs the 
+command `/deleteitem`.
+
+The general workflow of `/deleteitem` is as follows:
+1. User input is passed to `MoneyGoWhere`.
+1. `MoneyGoWhere` then creates a new `Command` object using the user input, whose constructor invokes 
+`Parser#formatArguments` method to extract the arguments for each flag into a `Map`.
+1. `Router#handleRoute` is then invoked to process the command and calls `Router#proRoute` which invokes 
+`Menu#deleteItem` method to run the `/deleteitem` command.
+1. Once the command runs, `Menu#deleteItem` checks if the list of items is empty. If empty, a message indicating that 
+there is nothing to be deleted using `MenuUi#printError` and control is given back to `MoneyGoWhere`.
+1. `Menu#deleteItem` invokes `DeleteItemValidation#validateFlags` to check if all the required flags have been given.
+    * If there are missing flags, a message indicating that the usage is invalid will be printed using 
+   `MenuUi#printError` and control is given back to `MoneyGoWhere`.
+1. `Menu#deleteItem` then invokes `DeleteItemValidation#validateCommand` which in turn, calls all of the following 
+validation method to check the arguments provided.
+    * `DeleteItemValidation#validateArgument` checks if the user input `String` is empty
+        * If the user input is empty, a message indicating that the input is empty is printed using `MenuUi#printError` 
+      and control is given back to `MoneyGoWhere`.
+    * `DeleteItemValidation#validateIndex` checks if the given index is valid and exists
+        * If the index is invalid, a message indicating the index does not exists is printed using `MenuUi#printError` 
+      and control is given back to `MoneyGoWhere`.
+1. `Menu#processDeleteItem` is then invoked, and it calls `Menu#removeItem` on the given index to delete it from the 
+list of items. Then, `Menu#save` is invoked to save the changes to the local storage file.
+1. `Router` object then calls `MenuUi#printCommandSuccess` to print a message indicating that the item has been 
+successfully added to the menu.
+
+
+##### Basic Mode [Delete an Item]()
```

Comment:
There is a strange hyperlink in the website. I think it is maybe a mistake in typing.